### DISCUSSION TITLE
document internal_filter

### DISCUSSION THREAD

**Discussion by TAW-V** [klaasnicolaas/home-assistant-glow#70] [Show and tell]
Hi,

Thanks for this great project.

I could fit everything in one small box of 36 x 61 x 25 mm with a bit of hot glue:

![Screenshot 2021-10-22 102453](https://user-images.githubusercontent.com/35802960/138421265-7d6dd1d5-3b86-4757-a54a-beed2c1ad5c5.jpg)
![Screenshot 2021-10-22 102510](https://user-images.githubusercontent.com/35802960/138421271-e77b463b-9185-4abc-85d5-c0a5ff079006.jpg)
![Screenshot 2021-10-22 102534](https://user-images.githubusercontent.com/35802960/138421275-de69eefa-036d-4163-a4c0-5a04e5ed72d3.jpg)
![Screenshot 2021-10-22 102602](https://user-images.githubusercontent.com/35802960/138421281-4c93f0c0-ac33-495c-9de1-b03e86558308.jpg)

link for this enclosure: https://nl.aliexpress.com/item/4001075657658.html?spm=a2g0s.12269583.0.0.54e439ffHmvZsU

As a suggestion i would tell on the main page to uncomment the internal_filter in the yaml file. It took me quite some time to figure out why i had strange sensor readings. But after uncommenting this line everything worked fine.

**Comment by klaasnicolaas** [comment]
`internal_filter` is not required by default and therefore also placed as a comment in the yaml code. I suspect you just needed it because your LED is not in line with the pulse led but sideways. What strange sensor reading did you have? Then I could write something about that in the FAQ.

Nice how you did it with the case!

**Comment by TAW-V** [comment]
i got a very high power reading most time. So i suspect a second pulse.

It is not shown on the photo, but i bent the LED ~80 degrees so that the led is perpendicular to the pulse light. That was necessary to detect the pulses.

I also had to reverse the LED as you suggested in the FAQ

**Reply by m-s-e** [reply]
my experience with case design is that it realy makes a big difference to get the sensor as close as possible to the meter LED, in every possible way!!
also ensure a tight contact between the case and the meter to minimise ambient light interference.
excuse my ignorance, but does the ESP you are using have an LED too? you can get amusing feedback type issues with interference light bouncing around in the box from the ESP LEDs!!!

**Reply by TAW-V** [reply]
Yes i had feedback problems of the other LEDs. So, I put some tape on the ESP LED and the other LED on the photo module to prevent feedback light.

Everything works great, just had to bend the LED to point it towards the meter and uncomment the internal_filter
